TimedArithmeticQuiz/src/components/QuestionHeader.jsx

```diff
@@ -0,0 +1,42 @@
+import React, { useEffect, useState } from "react";
+import { quizConfig } from "../data/quizConfig";
+import PropTypes from "prop-types";
+
+export const QuestionHeader = React.memo(function QuestionHeader({ questionNumber, onCountDownEnd }) {
+  const [secondsLeft, setSecondsLeft] = useState(quizConfig.perQuestionTimeInSecs);
+  const [countDownCompleted, setCountDownCompleted] = useState(false);
+
+  useEffect(() => {
+    if (countDownCompleted) {
+      setSecondsLeft(quizConfig.perQuestionTimeInSecs);
+      onCountDownEnd();
+      setCountDownCompleted(false);
+    }
+  },[countDownCompleted, onCountDownEnd]);
+
+  useEffect(() => {
+    setSecondsLeft(quizConfig.perQuestionTimeInSecs);
+    const intervalId = setInterval(() => {
+      setSecondsLeft(left => {
+        if(left === 1) setCountDownCompleted(true)
+        return left - 1;
+      });
+    }, 1000);
+
+    return () => {
```

Comment:
Why is this required?

Reply:
here I am returning this function to clear the interval. This is called when either the dependency array changes and useEffect is executed again or when the component is unmounted.
Here we want only one interval to run at a time and also ensure that there are no memory leaks, so this function acts as a cleaup function.